src/watchdog/observers/__init__.py

```diff
@@ -52,31 +52,37 @@
 
 """
 
+import typing as t
+import sys
 import warnings
-from watchdog.utils import platform
```

Comment:
Why such change?

Reply:
When I manually checked the sample code in `README.rst`, mypy complained that a mandatory `BaseObserver.__init__()` argument was not being filled. I couldn't find a way to show that *all* of the `BaseObserver` child classes override `.__init__()`, but found a way to convince mypy *enough* of the child classes do by telling mypy to only pay attention to the imports that matched the platform mypy is running on.

To do that, I replaced the `.is_<platform>()` calls with code that mypy understood. mypy supports the `sys.platform == ...` and `sys.platform.startswith(...)` syntax for marking code as platform-specific, and doing so here told mypy to skip checking ALL possible imports of `Observer` and instead only focus on the `Observer` imports that it actually could import.

When running on Windows, for example, mypy stopped caring about the `KqueueObserver` (which it couldn't import and introspect), which alleviated the complaint about the code in `README.rst`.

Hopefully that explanation makes sense! It's the only way I found to get the code in `README.rst` to pass so #840 can be closed.